colossus/src/main/scala/colossus/service/ServiceServer.scala

```diff
@@ -89,11 +88,14 @@ extends Controller[I,O](codec, ControllerConfig(50)) {
   private val requestBuffer = collection.mutable.Queue[SyncPromise]()
   private var numRequests = 0
 
-  def idleCheck(period: Duration) {
+  override def idleCheck(period: Duration) {
+    super.idleCheck(period)
```

Comment:
I totally agree, in this case it's a result of the idleCheck being declared on ConnectionHandler which is inherited all down the stack (and the fact that the ServiceClient/Controller hierarchy is inheritance based). Scala makes it extra fruity by allowing multiple inheritance so the end result depends on declaration order. To do what you suggest I think we'd need to repeat that hook pattern in a few places, it could get really messy.

Would be interested in @DanSimon 's views on this :)